spec/spec/features/user_index_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe 'User Index Page', type: :feature do
+  before do
+    visit users_path
+  end
+
+  it 'shows all users with their username and profile picture' do
+    User.all.each do |user|
+      expect(page).to have_content(user.name)
+      expect(page).to have_selector("img[src*='#{user.photo}']")
+    end
+  end
+
+  it 'shows the number of posts each user has written' do
+    User.all.each do |user|
+      expect(page).to have_content("Number of posts: #{user.posts.count}")
+    end
+  end
+
+  it 'redirects to user show page on clicking a user' do
+    user = User.first
+    # click_on user.name
```

Comment:
Kindly do well to take every commented piece of code from your spec files as it gives your codebase a messy look😉.

Reply:
- [ ] As mentioned by the previous reviewer, Kindly remove all commented out code from your code base. It make your code looks messy and unclean